lib/portage/dbapi/vartree.py

```diff
@@ -5086,21 +5086,18 @@ def onerror(e):
 
         emerge_log = self._emerge_log
 
-        # If we have any preserved libraries then autoclean
-        # is forced so that preserve-libs logic doesn't have
+        # We always autoclean now for the current package-case for simplicity.
+        # If it were conditional, we'd always need to do it when any preserved-libs,
+        # so that preserve-libs logic doesn't have
         # to account for the additional complexity of the
         # AUTOCLEAN=no mode.
-        autoclean = self.settings.get("AUTOCLEAN", "yes") == "yes" or preserve_paths
-
-        if autoclean:
-            emerge_log(_(" >>> AUTOCLEAN: %s") % (slot_atom,))
+        emerge_log(_(" >>> AUTOCLEAN: %s") % (slot_atom,))
 
         others_in_slot.append(self)  # self has just been merged
         for dblnk in list(others_in_slot):
             if dblnk is self:
                 continue
-            if not (autoclean or dblnk.mycpv == self.mycpv or reinstall_self):
```

Comment:
When changing to just:
```
	if not (dblnk.mycpv == self.mycpv or reinstall_self):
```

this broke the `lib/portage/tests/emerge/test_emerge_slot_abi.py` test, presumably because something overlapping was left, and normally we were testing with autoclean on.

@zmedico let me know if I should somehow adapt the test (and how?) if I haven't made the right fix here. It seems right to me to just _always_ autoclean here though like we discussed but I might be missing something (it feels wrong the test broke in the AUTOCLEAN=no case?)

Reply:
Right, we always want to cleanup after ourselves. The rest of this condition was for the "autoclean is False" case, which no longer exists, so the whole condition is unnecessary now.